sdk/endpoint/endpoint.go

```diff
@@ -199,7 +232,36 @@ func NewNSMEndpoint(ctx context.Context, configuration *common.NSConfiguration,
 	endpoint := &nsmEndpoint{
 		NsmConnection: nsmConnection,
 		service:       service,
+		registrations: nil,
+	}
+
+	// Default options
+	WithRegistrations(MakeRegistration(configuration))(endpoint)
```

Comment:
```suggestion
		registrations: []*Registration{MakeRegistration(configuration)},
	}
```

Reply:
It should be a bit more verbose since it should be an array of nested registration objects, but OK.